scripts/generate_ironside_ipc_request.py

```diff
@@ -0,0 +1,319 @@
+        "--base-address",
+        help="Base address for memory blob. This acts as the working area for the requests. All "
+        "data pointed to by the request is stored here. Any unused RAM available to the core that "
+        "is used to execute the IPC request can be given",
+        default=0x22000000,
+        type=lambda x: int(x, 0),
+    )
+
+    parser.add_argument(
+        "--output",
+        help="Output JSON filename (if not provided, outputs Intel HEX to stdout)",
+        type=argparse.FileType("wb"),
+        required=False,
+    )
+
+    subparsers = parser.add_subparsers(
+        dest="operation",
+        help="IPC operation to perform",
+        required=True,
+    )
+
+    import_parser = subparsers.add_parser(
+        "psa_import_key",
+        help="Invoke psa_import_key through the PSA service",
+    )
+
+    import_parser.add_argument(
+        "--psa-key-attrs",
+        help="PSA key attributes as hex string. This can be found by running the "
+        "generate_psa_key_attributes.py script",
+        type=str,
+        required=True,
+    )
+
+    import_parser.add_argument(
+        "--key-value",
+        help="Key value as hex string",
+        type=str,
+        required=True,
+    )
+
+    import_parser.add_argument(
+        "--key-id-address",
+        help="The address to store the resulting key_id to. The psa_import_key outputs the key_id "
+        " of the imported key. For persistent keys this will be the same value that was given as "
+        "input. It is still required to provide an address where this output key_id can be "
+        "written.",
+        type=lambda x: int(x, 0),
+        default=0x22001000,
+    )
+
+    random_parser = subparsers.add_parser(
+        "psa_generate_random",
+        help="Invoke psa_generate_random through the PSA service",
+    )
+
+    random_parser.add_argument(
+        "--output-address",
+        help="Base address for random output buffer (hex or decimal)",
+        type=lambda x: int(x, 0),
+        required=True,
+    )
+
+    random_parser.add_argument(
+        "--output-length",
+        help="Length of random output buffer (hex or decimal)",
+        type=lambda x: int(x, 0),
+        required=True,
+    )
+
+    args = parser.parse_args()
+
+    if args.operation == "psa_import_key":
+        ipc_request, request_data = generate_import_key_request(
+            args.psa_key_attrs,
+            args.key_value,
+            args.base_address,
+            args.key_id_address,
+            KEY_ID_LEN,
+        )
+    elif args.operation == "psa_generate_random":
+        ipc_request, request_data = generate_random_request(
+            args.output_address,
+            args.output_length,
+            args.base_address,
+        )
+
+    if args.output:
+        output_data = {
+            "ipc_request": binascii.hexlify(ipc_request).decode("utf-8").upper(),
+            "request_data": binascii.hexlify(request_data).decode("utf-8").upper(),
```

Comment:
There is also a `bytes.hex` that does a similar thing as `binascii.hexlify` but returns a string directly, so it should be possible to simplify:

```suggestion
            "ipc_request": ipc_request.hex().upper(),
            "request_data": request_data.hex().upper(),
```